src/common/core/librarycore.js

```diff
@@ -0,0 +1,785 @@
+/*globals define*/
+/*jshint node: true, browser: true*/
+
+/**
+ * @author kecso / https://github.com/kecso
+ */
+
+define([
+        'common/util/assert',
+        'common/core/tasync',
+        'common/core/constants',
+        'common/util/random'
+    ], function (ASSERT, TASYNC, CONSTANTS, RANDOM) {
+        'use strict';
+
+        var LibraryCore = function (innerCore, options) {
+            ASSERT(typeof options === 'object');
+            ASSERT(typeof options.globConf === 'object');
+            ASSERT(typeof options.logger !== 'undefined');
+
+            var logger = options.logger,
+                self = this,
+                key;
+
+            for (key in innerCore) {
+                this[key] = innerCore[key];
+            }
+
+            // We need to remove guid related functions so they will not appear on the public API.
+            delete this.getDataGuid;
+            delete this.getDeductedGuid;
+
+            logger.debug('initialized LibraryCore');
+
+            //<editor-fold=Helper Functions>
+            function getLibraryRootsInfo(node) {
+                var allMetaNodes = self.getAllMetaNodes(node),
+                    libraryRoots = {},
+                    path,
+                    roots,
+                    i,
+                    name;
+
+                for (path in allMetaNodes) {
+                    roots = getAllLibraryRoots(allMetaNodes[path]);
+                    for (i = 0; i < roots.length; i += 1) {
+                        name = getLibraryName(roots[i]) + self.getAttribute(roots[i], 'name');
```

Comment:
If we use libraries the name has a special meaning, maybe we should consider not being able to remove it from the UI at least.